.github/workflows/build-toolchain.yml

```diff
@@ -301,208 +329,191 @@ jobs:
+              "triple_no_api_level": "aarch64-unknown-windows-msvc",
+              "cc": "cl",
+              "cflags": "${{ steps.context.outputs.WINDOWS_CMAKE_C_FLAGS }}",
+              "cxx": "cl",
+              "cxxflags": "${{ steps.context.outputs.WINDOWS_CMAKE_CXX_FLAGS }}",
+              "swiftflags": "${{ steps.context.outputs.WINDOWS_CMAKE_Swift_FLAGS }}",
+              "cmake_linker_flags": "-D CMAKE_EXE_LINKER_FLAGS=\"${{ steps.context.outputs.WINDOWS_CMAKE_EXE_LINKER_FLAGS }}\" -D CMAKE_SHARED_LINKER_FLAGS=\"${{ steps.context.outputs.WINDOWS_CMAKE_EXE_LINKER_FLAGS }}\"",
+              "llvm_flags": "",
+              "extra_flags": "-D CMAKE_SYSTEM_PROCESSOR=ARM64 -D CMAKE_MT=mt"
+            }
+          ANDROID_ARM64: >-
             {
-              "include": [
-                {
-                  "arch": "amd64",
-                  "compiler_target": "x86_64-unknown-windows-msvc",
-                  "os": "Windows",
-                  "cc": "cl",
-                  "cflags": "${{ steps.context.outputs.WINDOWS_CMAKE_C_FLAGS }}",
-                  "cxx": "cl",
-                  "cxxflags": "${{ steps.context.outputs.WINDOWS_CMAKE_CXX_FLAGS }}",
-                  "swiftflags": "${{ steps.context.outputs.WINDOWS_CMAKE_Swift_FLAGS }}",
-                  "extra_flags": "-D CMAKE_SYSTEM_PROCESSOR=AMD64 -D CMAKE_MT=mt"
-                },
-                {
-                  "arch": "arm64",
-                  "compiler_target": "aarch64-unknown-windows-msvc",
-                  "os": "Windows",
-                  "cc": "cl",
-                  "cflags": "${{ steps.context.outputs.WINDOWS_CMAKE_C_FLAGS }}",
-                  "cxx": "cl",
-                  "cxxflags": "${{ steps.context.outputs.WINDOWS_CMAKE_CXX_FLAGS }}",
-                  "swiftflags": "${{ steps.context.outputs.WINDOWS_CMAKE_Swift_FLAGS }}",
-                  "extra_flags": "-D CMAKE_SYSTEM_PROCESSOR=ARM64 -D CMAKE_MT=mt"
-                }
-              ]
+              "arch": "arm64",
+              "cpu": "aarch64",
+              "os": "Android",
+              "platform": "Android",
+              "triple": "aarch64-unknown-linux-android${{ steps.context.outputs.ANDROID_API_LEVEL }}",
+              "triple_no_api_level": "aarch64-unknown-linux-android",
+              "cc": "clang",
+              "cflags": "${{ steps.context.outputs.ANDROID_CMAKE_C_FLAGS }}",
+              "cxx": "clang++",
+              "cxxflags": "${{ steps.context.outputs.ANDROID_CMAKE_CXX_FLAGS }}",
+              "swiftflags": "-sdk $NDKPATH/toolchains/llvm/prebuilt/windows-x86_64/sysroot -Xclang-linker -target -Xclang-linker aarch64-unknown-linux-android${{ steps.context.outputs.ANDROID_API_LEVEL }} -Xclang-linker --sysroot -Xclang-linker $NDKPATH/toolchains/llvm/prebuilt/windows-x86_64/sysroot -Xclang-linker -resource-dir -Xclang-linker $NDKPATH/toolchains/llvm/prebuilt/windows-x86_64/lib/clang/17 -L ${GithubWorkspace}/BinaryCache/swift/lib/swift/android -g",
```

Comment:
I think a potential issue might be with declaring all the flags here be that there are may be flags that need to be set conditionally. For example, there's no NDK for Windows/arm64. We'd need to elide the `NDKPATH` from the flags conditionally if it's for Windows/arm64 and we need drop the flags like `-DCMAKE_ANDROID_NDK` and `-DSWIFT_ANDROID_NDK_PATH` instead of passing empty values, or they will err. For example, around `swift-toolchain.yml:1659` in [the PR](https://github.com/compnerd/swift-build/pull/847/files). Could we avoid that, as in would it be a real issue?

Reply:
When I was working on getting the `sdk` job to build on Mac (which I have since abandoned), I encountered a similar issue. I solved it by making the NDK flags conditional to `"${{ matrix.os }}" -eq "Android"` in a preliminary `setup-config` step. See this [branch](https://github.com/compnerd/swift-build/commit/e5bd0a6b0cf941a93d7be2fc69379683fb40d094#diff-cd548d9f13bbeda6b71038430c870dd197f12e4006bdfdb5f87c6c7f442caf81R2008)

This should not be a problem if I can adapt this WIP branch on top of this PR to only add the NDK-related arguments for Android. This will also not be part of any build matrix with Windows Arm64 as the builder. Does that make sense?

Reply:
It sounds like there's a way to handle it, sure :)